JiscSD/octopus
# Sending publications to PubRouter - (infra)

The purpose of this PR was to create a lambda that sends a publication's pdf to pubrouter once the pdf has been generated and saved in an s3 bucket. This was previously [OCT-577](https://github.com/JiscSD/octopus/pull/425) but I've had to open a new branch as rebasing it with main has caused a few problems. 

An overview of the architecture/process is as follows (in bold are the parts included in this PR):

A publication is created ⏩ A PDF is generated and stored in a s3 bucket ⏩ **The Lambda function is triggered and a delivery with the publication's metadata and pdf link to PubRouter is attempted
If this fails then a delivery is reattempted ⏩ If this fails again the failure is logged on a slack channel**

---
### How to Test: 

- After deploying the terraform to int, you can either upload a publication’s pdf to the s3 bucket (quickest way) or create a publication on int 
- This will trigger the lambda, and the initial pubrouter response will be in the lambda's cloudwatch logs. If it fails there will be a slack notification in the failed-pubrouter-deliveries channel with any errors and issues with the publication
- You can also log into the pubrouter dashboard to see a history of all of the publications sent to pubRouter along with the errors 

---

### Release considerations: 

There are some new secrets that are coming from SSM parameter store. We will need to add the production versions of these to SSM before deploying to prod.

---

### Acceptance Criteria:

As per [OCT-577](https://jiscdev.atlassian.net/jira/software/projects/OCT/boards/836?selectedIssue=OCT-577) - Please check the comments for updates on the metadata following a conversation I've had with Tim

[OCT-577]: https://jiscdev.atlassian.net/browse/OCT-577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
